src/Spice86.Core/Emulator/InterruptHandlers/Dos/DosInt21Handler.cs

```
@@ -89,6 +103,54 @@ public class DosInt21Handler : InterruptHandler {
         _dispatchTable.Add(0x62, new Callback(0x62, GetPspAddress));
     }
 
+    public void TerminateProgram() {
+        if (_loggerService.IsEnabled(LogEventLevel.Verbose)) {
+            _loggerService.Verbose("Program requested exit via INT2F AH=0x00 (Terminate Program). Terminating now");
+        }
+        _cpu.IsRunning = false;
+    }
+
+    public void BufferedInput() {
+        
+    }
+
+    /// <summary>
+    /// Waits for a character from standard input, echoes it to standard output, and returns the character in the AL register. <br/>
+    /// Echoes on standard output.
+    /// </summary>
+    public void CharacterInputWithEcho() {
+        // Standard input is not implemented, we return no character ready.
```

Please log it, so it isn't just silently ignored.